cdap-system-app-api/src/main/java/io/cdap/cdap/api/service/http/SystemHttpServiceContext.java

```diff
@@ -61,4 +61,15 @@
    */
   Map<String, String> evaluateMacros(String namespace, Map<String, String> properties,
                                      MacroEvaluator evaluator, MacroParserOptions options) throws InvalidMacroException;
+
+  /**
+   * Get preferences for the given entity
+   *
+   * @param entityId the id of the entity to fetch preferences for
+   * @param resolved true if resolved properties are desired.
+   * @return Map containing Preferences keys and values.
+   */
+  default Map<String, String> getPreferences(EntityId entityId, boolean resolved) {
+    throw new IllegalStateException("Not implemented");
```

Comment:
We usually use UnsupportedOperationException for this.

Reply:
I'll change this exception.